### PR TITLE
.proto enum variable named the same as enumerator causes compilation …

### PR DESCRIPTION
**Pull request by mellster2012**
…error #299

### CODE REVIEW

**Comment by oldergod** [comment]
Would you be able to write a test as well?

**Comment by mellster2012** [comment]
Sure will do.

**Comment by mellster2012** [comment]
@oldergod @jrodbx Test added.

File: wire-java-generator/src/test/java/com/squareup/wire/java/JavaGeneratorTest.java [hunk 1]
```diff
@@ -444,4 +444,19 @@ public void enclosingTypeIsNotMessage() {
...
   }
+
+  @Test
+  public void generateTypeUsesPackageNameOnFieldAndClassNameClash() throws Exception {
+    RepoBuilder repoBuilder = new RepoBuilder()
+	    .add("person.proto", ""
+	        + "package common.proto;\n"
+	        + "enum Gender {\n"
+	        + "Gender_Male = 0;\n"
+	        + "Gender_Female = 1;\n"
```
Comment (oldergod) [inline wire-java-generator/src/test/java/com/squareup/wire/java/JavaGeneratorTest.java]: 2 spaces indents. Same for line 458.

File: wire-java-generator/src/test/java/com/squareup/wire/java/JavaGeneratorTest.java [hunk 2]
```diff
@@ -444,4 +444,19 @@ public void enclosingTypeIsNotMessage() {
...
+	        + "}\n"
+	        + "message Person {\n"
+	        + "optional Gender Gender = 1;\n"
+	        + "}\n");
+	    assertThat(repoBuilder.generateCode("common.proto.Person").contains("public final Gender common_proto_Gender;"));
```
Comment (oldergod) [inline wire-java-generator/src/test/java/com/squareup/wire/java/JavaGeneratorTest.java]: The assertion is wrong.
`assertThat(repoBuilder.generateCode("common.proto.Person").contains("public final Gender common_proto_Gender;"));`
should be
`assertThat(repoBuilder.generateCode("common.proto.Person")).contains("public final Gender common_proto_Gender;");`
you could have noticed it via the IDE warning, or by running the test without your fix in JavaGenerator.

also, add a new line before `.contains` for we authorize up to 100 char / line.

**Comment by oldergod** [comment]
Thank you, our company also require you to sign [our CLA](https://docs.google.com/forms/d/e/1FAIpQLSeRVQ35-gq2vdSxD1kdh7CJwRdjmUA0EZ9gRXaWYoUeKPZEQQ/viewform?formkey=dDViT2xzUHAwRkI3X3k5Z0lQM091OGc6MQ&ndplr=1).

**Comment by mellster2012** [comment]
@oldergod Thanks for catching the wrong assertion - all updates made and CLA signed.

**Pull request merged**